### PR TITLE
[pkp-lib] #6071

### PR DESCRIPTION
**Pull request by defstat**

### CODE REVIEW

**Review by bozana — COMMENTED** [review]
Hi @defstat, just two questions. One is connected with the questions in the plugin code...
Thanks a lot!!!

File: classes/publication/PKPPublicationDAO.inc.php [hunk 1]
```diff
@@ -315,4 +315,55 @@ public function deleteAllPubIds($contextId, $pubIdType) {
...
+	 * Find publication by querying settings.
+	 * @param $settingName string
+	 * @param $settingValue mixed
+	 * @param $contextId int optional
+	 * @return array Publication.
+	 */
+	public function getIdsBySetting($settingName, $settingValue, $contextId) {
+		$q = Capsule::table('publications as p')
+			->join('publication_settings as ps', 'p.publication_id', '=', 'ps.publication_id')
+			->join('submissions as s', 'p.submission_id', '=', 's.submission_id')
```
[the remark this answers is not in the document]
Comment (bozana) [reply classes/publication/PKPPublicationDAO.inc.php:360]: Ah, I see, this is a functio to generally get publications with any setting... Sorry!

File: classes/submission/PKPSubmissionDAO.inc.php [hunk 2]
```diff
@@ -312,4 +313,23 @@ function getSortDirection($sortOption) {
 		list($sortBy, $sortDir) = explode("-", $sortOption);
 		return $sortDir;
 	}
+
+	/**
+	 * Find submission by querying settings.
```
Comment (bozana) [inline classes/submission/PKPSubmissionDAO.inc.php:318]: Find submission IDs

File: classes/submission/PKPSubmissionDAO.inc.php [hunk 3]
```diff
@@ -312,4 +313,23 @@ function getSortDirection($sortOption) {
...
+	 * @param $settingName string
+	 * @param $settingValue mixed
+	 * @param $contextId int optional
```
Comment (bozana) [inline classes/submission/PKPSubmissionDAO.inc.php:321]: contextID is not optional

File: classes/submission/PKPSubmissionDAO.inc.php [hunk 4]
```diff
@@ -312,4 +313,23 @@ function getSortDirection($sortOption) {
...
+	 * @return array Publication.
```
Comment (bozana) [inline classes/submission/PKPSubmissionDAO.inc.php:322]: array of submisison IDs

File: classes/publication/PKPPublicationDAO.inc.php [hunk 5]
```diff
@@ -315,4 +315,55 @@ public function deleteAllPubIds($contextId, $pubIdType) {
...
+	/**
+	 * Retrieve Publication by pubId
+	 * @param $pubIdType string One of the NLM pub-id-type values or
+	 * 'other::something' if not part of the official NLM list
+	 * (see <http://dtd.nlm.nih.gov/publishing/tag-library/n-4zh0.html>).
+	 * @param $pubId string
+	 * @param $contextId int
+	 * @return Publication|null
+	 */
+	public function getByPubId($pubIdType, $pubId, $contextId) {
```
[the remark this answers is not in the document]
Comment (defstat) [reply classes/publication/PKPPublicationDAO.inc.php:328]: It did not. I removed it

File: classes/submission/PKPSubmissionDAO.inc.php (hunk 3 above)
Comment (defstat) [reply classes/submission/PKPSubmissionDAO.inc.php:321]: Fixed

File: classes/submission/PKPSubmissionDAO.inc.php (hunk 4 above)
Comment (defstat) [reply classes/submission/PKPSubmissionDAO.inc.php:322]: Fixed

File: classes/submission/PKPSubmissionDAO.inc.php (hunk 2 above)
Comment (defstat) [reply classes/submission/PKPSubmissionDAO.inc.php]: Fixed

**Pull request merged**